atomic_reactor/plugins/post_group_manifests.py

```diff
@@ -0,0 +1,212 @@
+        :param registries: dict, keys are docker registries, values are dicts containing
+                           per-registry parameters.
+                           Params:
+                            * "secret" optional string - path to the secret, which stores
+                              login and password for remote registry
+        :param group: bool, if true, create a manifest list; otherwise only add tags to
+                      amd64 image manifest
+        :param goarch: dict, keys are platform, values are go language platform names
+        """
+        # call parent constructor
+        super(GroupManifestsPlugin, self).__init__(tasker, workflow)
+        self.group = True # group
+        self.goarch = goarch or {}
+        self.registries = registries
+        self.worker_registries = {}
+
+    def submit_manifest_list(self, registry, registry_conf, manifest_list_spec):
+        docker_secret_path = registry_conf.get('secret', None)
+#        dockercfg = "{0}/.dockercfg".format(registry_conf.get('secret', None))
+        
+        with NamedTemporaryFile(prefix='manifest-list', suffix=".yml", mode='w') as fp:
+            yaml.dump(manifest_list_spec, stream=fp)
+            fp.flush()
+            self.log.debug("Wrote to file %s with config %s", fp.name, docker_secret_path)
+
+            cmd = ['manifest-tool', '--docker-cfg=%s' % docker_secret_path,
+                   'push', 'from-spec', fp.name]
+            subprocess.check_call(cmd, env={'HOME': docker_secret_path})
+            self.log.info("Manifest list submitted for %s", registry)
+
+    def get_grouped_manifests(self):
+        grouped_manifests = []
+        for registry, registry_conf in self.registries.items():
+            manifest_list_spec = {}
+            manifest_list_spec['manifests'] = []
+            all_annotations = self.workflow.build_result.annotations['worker-builds']
+            for platform in all_annotations:
+                worker_image = all_annotations[platform]['digests'][0]
+                digest = worker_image['digest']
+                repository = worker_image['repository']
+                arch_entry = {
+                    'image': '{0}/{1}@{2}'.format(registry, repository, digest),
+                    'platform': {
+                        'os': 'linux',
+                        'architecture': self.goarch.get(platform, platform)
+                    }
+                }
+                manifest_list_spec['manifests'].append(arch_entry)
+
+            manifest_list_spec['tags'] = [image.tag for image in self.workflow.tag_conf.images]
+            # use a unique image tag because manifest_list doesn't like digests
+            registry_image = self.workflow.tag_conf.unique_images[0]
+            registry_image.registry = registry
+            manifest_list_spec['image'] = registry_image.to_str()
+            self.log.info("Submitting manifest-list %s", manifest_list_spec)
```

Comment:
The manifest-tool documentation calls this input file a 'spec'. Let's call it that in the log message too, to avoid confusion with the actual 'manifest list' that manifest-tool creates as its output.